samcli/commands/sync/command.py

```diff
@@ -403,10 +410,39 @@ def execute_code_sync(
         else set(get_all_resource_ids(stacks))
     )
 
+    sync_function_event_mapper = {
+        AutoDependencyLayerParentSyncFlow: "AutoDependencyLayerParent",
+        ImageFunctionSyncFlow: "ImageFunction",
+        ZipFunctionSyncFlow: "ZipFunction",
+    }
+    sync_misc_event_mapper = {
+        "Alias": "AliasVersion",
+        "Function Layer Reference Sync": "FunctionLayerReference",
+        "HttpApi": "HttpApi",
+        "RestApi": "RestApi",
+        "StepFunctions": "StepFunctions",
+    }
+
     for resource_id in sync_flow_resource_ids:
         sync_flow = factory.create_sync_flow(resource_id)
         if sync_flow:
             executor.add_sync_flow(sync_flow)
+            # Track the type of sync flow
+            name = sync_flow.log_name
```

Comment:
If we track the class name of sync flows as event value, would that make this block code simpler?

Reply:
I tried that initially, but I can definitely take a deeper look into doing that! I think I might have a new idea for how that can be simplified